src/main/java/com/omnisci/jdbc/OmniSciPreparedStatement.java

```diff
@@ -375,11 +380,12 @@ public void addBatch() throws SQLException {
           } catch (TException ex) {
             throw new SQLException(ex.toString());
           }
+          
           for (int i = 0; i < fieldsOrder.length; i++) {
-            fieldsOrder[i] = listOfColumns.indexOf(listOfFields[i].toLowerCase());
+            fieldsOrder[i] = listOfFieldsLowerCase.indexOf(listOfColumns.get(i).toLowerCase());
```

Comment:
Here and a couple lines below: `listOfColumns` is already lowercase, right? No need to cast it again.

Reply:
@Smyatkin-Maxim  - Indeed, those two calls are redundant. Updated code.